cmd/mdatagen/metrics_v2.tmpl

```diff
@@ -101,6 +101,7 @@ func newMetric{{ $name.Render }}(settings MetricSettings) metric{{ $name.Render
 // MetricsBuilder provides an interface for scrapers to report metrics while taking care of all the transformations
 // required to produce metric representation defined in metadata and user settings.
 type MetricsBuilder struct {
+	data                     *pdata.Metrics  // data buffer for generated metric.
```

Comment:
Each field `metric{{ $name.Render }}` is already a buffer. So I don't think we need this. We can create `pdata.Metrics` in the `Emit` call (sorry the the confusing description in the issue that I added)

Reply:
Please take a look at the update implementation @dmitryax, I followed your suggestions and think it's much cleaner